cpu/stm32_common/stmclk_common.c

```diff
@@ -47,35 +47,35 @@ void stmclk_disable_hsi(void)
 {
     /* we only disable the HSI clock if not used as input for the PLL and if
      * not used directly as system clock */
-#if CLOCK_HSE
-    if ((RCC->CFGR & RCC_CFGR_SWS) != RCC_CFGR_SWS_HSI) {
-        RCC->CR &= ~(RCC_CR_HSION);
+    if (CLOCK_HSE) {
```

Comment:
though I know, we don't use clang for ARM (currently/yet) but newer GCC might also be not happy?